R/mod_popExp_fct_scatterplot.R

```diff
@@ -132,7 +132,8 @@ app_scatterplot <- function(data, yvar, xvar, week_x, value_x, week_y, value_y,
         dplyr::select(USUBJID, PARAMCD, value_y, one_of(color, separate)) %>%
         tidyr::pivot_wider(names_from = PARAMCD, values_from = value_y) %>%
         tidyr::unnest(yvar) %>% # if their are more than 1 AVAL per Patient, per Visit
-        select_if(~!all(is.na(.))) # remove NA cols
+        dplyr::mutate(across(where(function(x) all(is.na(x))), ~ "NA" )) #%>% # Convert NA cols
+        # select_if(~!all(is.na(.))) # remove NA cols
```

Comment:
What is the purpose of `select_if(~!all(is.na(.)))`? The solution I see here would be achieved by just commenting that line out completely. I'm assuming it is meant to handle a situation that can arise from `pivot_wider()`. Maybe the mutate should only be applied to `color` and `separate`.

Reply:
I see that it is not equivalent to commenting out the last line. Specifically the color grouping will not work if the field is completely missing (i.e. `all(is.na(.))`)